app/services/reindex_service.rb

```diff
@@ -9,16 +9,33 @@ def initialize(solr:)
     @solr = solr
   end
 
-  # Switches out the ActiveFedora::SolrService connection to a
-  # a solr server that you specify and then reindexes your
-  # repository content for that server. It then restores the original
-  # connection
-  #
+  # Reindexes your repository content to the
+  # alternate solr server.
   # @return [Boolean] the reindex success
   def reindex
+    result = nil
+    connected_to_alternate_solr do
+      result = ActiveFedora::Base.reindex_everything
+    end
+    result
+  end
+
+  # This method does:
+  #   1. Switch out the ActiveFedora::SolrService
+  #      connection to the alternate solr server
+  #   2. Do some work
+  #   3. Restore the original connection
+  def connected_to_alternate_solr
     ActiveFedora::SolrService.instance.conn = RSolr.connect url: @solr
-    result = ActiveFedora::Base.reindex_everything
+    yield
```

Comment:
:+1: I still have some doubts about this method and its effects on the global solr service; e.g. I think it's not thread safe if this service is ever called from an actual running server (rather than a rake task). `yield` seems like a nice improvement, though.